app/controllers/spree/admin/overview_controller_decorator.rb

```diff
@@ -1,26 +1,73 @@
 Spree::Admin::OverviewController.class_eval do
   def index
-    # TODO was sorted with is_distributor DESC as well, not sure why or how we want ot sort this now
-    @enterprises = Enterprise.managed_by(spree_current_user).order('is_primary_producer ASC, name')
+    @enterprises = Enterprise
+      .managed_by(spree_current_user)
+      .order('is_primary_producer ASC, name')
     @product_count = Spree::Product.active.managed_by(spree_current_user).count
     @order_cycle_count = OrderCycle.active.managed_by(spree_current_user).count
 
-    unspecified = spree_current_user.owned_enterprises.where(sells: 'unspecified')
-    outside_referral = !URI(request.referer.to_s).path.match(/^\/admin/)
+    if first_access
+      redirect_to enterprises_path
+    else
+      render dashboard_view
+    end
+  end
 
-    if OpenFoodNetwork::Permissions.new(spree_current_user).manages_one_enterprise? && !spree_current_user.admin?
+  private
+
+  # Checks whether the user is accessing the admin for the first time
+  #
+  # @return [Boolean]
+  def first_access
+    outside_referral && unspecified_sales?
+  end
+
+  # Checks whether the request comes from another admin page or not
+  #
+  # @return [Boolean]
+  def outside_referral
+    !URI(request.referer.to_s).path.match(%r{/admin})
+  end
+
+  # Checks if any of the enterprises owned by the current user doesn't sell to
+  # anyone known yet.
```

Comment:
This could perhaps more appropriately be called `incomplete_enterprise_registration?`.

And the comment could be something like: `Checks that all of the enterprises owned by the current user have a 'sells' property specified, which indicates that the registration process has been completed`